### DISCUSSION TITLE
How to add custom preview options in WP 6.7

### DISCUSSION THREAD

**Discussion by justintadlock** [WordPress/developer-blog-content#315] [Topic Ideas]
There's a new Preview Options API coming in WordPress 6.7: https://github.com/WordPress/gutenberg/pull/64644#top

It allows you to add custom options to the preview dropdown in the editor:

![image](https://github.com/user-attachments/assets/3752e5b4-4109-483c-8492-5602ead8d4c8)

What we need is a step-by-step walkthrough of how to implement custom preview options. The API is just for adding the preview button in the dropdown, but the `onClick()` method (what happens when the button is clicked) is custom. We could share several different ideas on what's possible and what it could be used for.

**Comment by bph** [comment]
This topic was approved in Oct Editorial Group meeting. Discussion continues on the issue 
- #328

**Comment by bph** [comment]
Published [How to add custom entries to the editor Preview dropdown](https://developer.wordpress.org/news/2026/02/how-to-add-custom-entries-to-the-editor-preview-dropdown/)
